File: data-pipeline/druid-events-validator/src/main/java/org/ekstep/ep/samza/domain/Event.java
package org.ekstep.ep.samza.domain;

import org.ekstep.ep.samza.events.domain.Events;
import org.ekstep.ep.samza.task.DruidEventsValidatorConfig;

import java.util.HashMap;
import java.util.Map;

public class Event extends Events {


    public Event(Map<String, Object> map) {
        super(map);
    }

    public void markSuccess() {
        telemetry.addFieldIfAbsent("flags", new HashMap<String, Boolean>());
        telemetry.add("flags.dv_processed", true);
        telemetry.add("type", "events");
    }

    public void markFailure(String error, DruidEventsValidatorConfig config) {
        telemetry.addFieldIfAbsent("flags", new HashMap<String, Boolean>());
        telemetry.add("flags.dv_processed", false);

        telemetry.addFieldIfAbsent("metadata", new HashMap<String, Object>());
        if (null != error) {
            telemetry.add("metadata.dv_error", error);
            telemetry.add("metadata.src", config.jobName());
        }

    }

    public boolean isSummaryEvent() {
        return eid() != null && eid().equals("ME_WORKFLOW_SUMMARY");
    }

    public boolean isSearchEvent() {
        return "SEARCH".equalsIgnoreCase(eid());
    }

    public boolean isLogEvent() {
        return "LOG".equalsIgnoreCase(eid());
    }
}

File: data-pipeline-flink/druid-events-validator/src/main/scala/org/sunbird/dp/domain/Event.scala
package org.sunbird.dp.domain

import java.util

class Event(eventMap: util.Map[String, AnyRef], partition: Integer) extends Events(eventMap, partition) {

  private val jobName = "DruidValidator"

  def markValidationSuccess(): Unit = {
    this.updateFlags("dv_processed", true)
  }

  def markSkippedValidation(): Unit = {
    this.updateFlags("dv_validation_skipped", true)
  }

  def markSkippedDedup(): Unit = {
    this.updateFlags("dv_dedup_skipped", true)
  }

  def markValidationFailure(errorMsg: String): Unit = {
    this.updateFlags("dv_processed", false)
    this.updateFlags("dv_validation_failed", true)
    telemetry.addFieldIfAbsent("metadata", new util.HashMap[String, AnyRef])
    if (null != errorMsg) {
      telemetry.add("metadata.dv_error", errorMsg)
      telemetry.add("metadata.src", jobName)
    }
  }

  def isSummaryEvent: Boolean = "ME_WORKFLOW_SUMMARY".equalsIgnoreCase(eid)

  def isSearchEvent: Boolean = "SEARCH".equalsIgnoreCase(eid)

  def isLogEvent: Boolean = "LOG".equalsIgnoreCase(eid)

  def isErrorEvent: Boolean = "ERROR".equalsIgnoreCase(eid)

}
